minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/patch/EntrypointPatch.java

```diff
@@ -510,6 +567,17 @@ public void process(FabricLauncher launcher, Function<String, ClassNode> classSo
 					break;
 				}
 			}
+
+			// TODO: better handling of run directory for pre-classic
+			if (!patched && gameMethod != gameConstructor) {
+				ListIterator<AbstractInsnNode> it = gameMethod.instructions.iterator();
+
+				it.add(new InsnNode(Opcodes.ACONST_NULL));
```

Comment:
Regarding the todo, what actually is the run directory in pre-classic? If it's the working directory, then this can be `new File(".")` which would be better than `null`.

Reply:
Ah I see